### PR TITLE
release-20.2: keys: avoid creating invalid utf-8 sequences in logs

### PR DESCRIPTION
**Pull request by knz**
Backport 1/1 commits from #53973.

/cc @cockroachdb/release

---

### CODE REVIEW

**Comment by otan** [comment]
LGTM, but subject to release blockers backport approval

**Comment by knz** [comment]
cc @nvanbenschoten TL in this area

**Review by nvb — APPROVED** [review]

**Comment by otan** [comment]
i thought you had to wait for release to sign off if it isn't a release blocker before merging

**Comment by knz** [comment]
> i thought you had to wait for release to sign off if it isn't a release blocker before merging

oops I was not aware. I'm going to review the process. Thanks for the heads up.

**Pull request merged**
